lambdas/service/app.py

```diff
@@ -1043,7 +1047,10 @@ def repository_search(entity_type: str, item_id: Optional[str]) -> JSON:
 catalog_param_spec = params.query(
     'catalog',
     schema.optional(schema.with_default(app.catalog,
-                                        type_=schema.enum(*config.catalogs))),
+                                        type_={
+                                            **schema.pattern(f'^{IndexName.catalog_name_re.pattern}$'),
```

Comment:
When a pattern is embedded in another one, it needs to be enclosed in parentheses. Be sure to understand why.